src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserCredentialsException;
+import core.basesyntax.exception.InvalidUserObjectException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (validateUserCredentials(user)) {
+            return storageDao.add(user);
+        }
         return null;
     }
+
+    public boolean validateUserCredentials(User user) {
+        if (user == null) {
+            throw new InvalidUserObjectException("User object cannot be null");
+        }
+        if (user.getAge() <= 0) {
+            throw new InvalidUserCredentialsException("User age cannot be negative or zero");
+        }
+        if (user.getAge() < 18) {
+            throw new InvalidUserCredentialsException("User age cannot be less than 18");
+        }
+        validateUserEmail(user);
+        validateUserPassword(user);
+        return true;
+    }
+
+    public void validateUserEmail(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidUserCredentialsException("User login cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserCredentialsException("User login length cannot be less than 6");
```

Comment:
Consider moving the minimum login length to a constant field for better maintainability and readability. This aligns with the checklist recommendation to move hardcoded values to constant fields.